### DISCUSSION TITLE
Welcome to nhl-api-py Discussions!

### DISCUSSION THREAD

**Discussion by coreyjs** [coreyjs/nhl-api-py#6] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by coreyjs** [comment]
Test test is this thing on?

**Comment by Btibert3** [comment]
I just came across this project.  Thank you for the incredible amount of work you put into building this out.

**Reply by coreyjs** [reply]
Thank you!  Its been a fun little side project.

**Comment by Glumdar** [comment]
Thank you for the work on making this available.  I was trying to working directly with the APIs and was going nuts.

I'm new to this and may be using the wrong terminology. I noticed that in the method "teams_info" using date "2021-05-01" it will fail.  I poked around a bit and noticed that for the code in it 
```
team = {
"conference": {"abbr": i["conferenceAbbrev"], "name": i["conferenceName"]},
"division": {"abbr": i["divisionAbbrev"], "name": i["divisionName"]},
"name": i["teamName"]["default"],
"common_name": i["teamCommonName"]["default"],
"abbr": i["teamAbbrev"]["default"],
logo": i["teamLogo"],
}
```
it fails because the results do not have a **conferenceAbbrev** or **divisionAbbrev** field.  That was during covid so I guess they organized things a bit different that year.  Or I'm doing something wrong :-p

Just thought I would share.

-Alan

**Reply by coreyjs** [reply]
@Glumdar good catch!  I can make this method a little safer, this is a tricky edge case.

**Comment by benjaminmccallum** [comment]
Hi there, is there betting odds data somewhere? I thought I saw it in the schedule endpoint but it appears to no longer be there.

**Reply by coreyjs** [reply]
There def was this data on games, let me dig around and see what happened.

**Reply by coreyjs** [reply]
They are visible on the daily schedule:  `client.schedule.daily_schedule(date="2025-10-13")` keyed by `oddsPartners`.

I dont see them in the payload of individual games or Game Center, it just seems to be this daily schedule one.

EDIT:
And for some reason they no longer show odds or spreads. This changed on their end.  Im not sure when.

**Comment by Btibert3** [comment]
FWIW, there are some betting data tagged via the ESPN APIs, I believe.

**Comment by bananasfourscale** [comment]
Hi, has anyone noticed rate limiting while pulling from the API? I never noticed it in the previous 4 season that I've pulled from the two API's over that time, but now as of the last week or so I'm consistently getting request bans if I query too quickly. I've not been able to find what the limit is in order to build a high band limiter in my own personal code.

**Reply by coreyjs** [reply]
I have noticed this also. It's annoying. I usually have to put in a bunch of 1 second sleep calls if I'm doing any aggregation type work which makes everything take longer.

**Reply by bananasfourscale** [reply]
Ah, well at least its not just me. My entire project is aggregates so its going to be interesting. I was already thinking of just saving off all the data I query and making a local database anyway so maybe this will push me to do that. Thanks for the reply.
